### ISSUE TITLE
Display area fills on any zoom level if the feature is big enough for it

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#2856]
We have a feature already to hide area objects if they're very small for a current zoom level. E.g. a building or a patch of forest becomes near dot size at some zoom and its no longer feasible to render it.

However why hard-limit display of big area features?
E.g. hospital grounds are displayed starting from zoom level 14 at the moment.
But many hospital are so big, that even on zoom level 10 they may occupy more screen space than e.g. adjacent patches of forest which are displayed still.
The same goes to industrial areas, buildings, schools etc.

Here is a preview how the map would look like if we display area features on any zoom level if they're big enough for it:
![no-limit-areas](https://user-images.githubusercontent.com/18434508/176224373-3c933aa7-8905-4cc2-b67d-97af8ce7043f.png)

The map becomes more colorful and its easy to see at lower zooms where are e.g. industrial or educational clusters are located.

Are there any use cases when we don't want this behavior?

ATM the hatching looks ugly for zooms <12. 
Also we might want not to display residential/commercial/etc landuse #2184 at UPPER zoom levels (if we ever add these features)?

**Comment by biodranik** [comment]
Is it only about drawing some land areas earlier? Need to see it visually in different cities, probably it should be ok.

**Comment by PetbkA** [comment]
I wanted to create a separate issue but I guess this one has the same roots.

**OS:** LineageOS 17.1 (Android 10)
**Device:** Xiaomi Redmi Note 9 Pro
**OM:** version 2022.06.29-3-FDroid

**Steps:**
1. Go to [Ferry](https://www.openstreetmap.org/node/1309605858)  at 59.941456, 30.312279 coordinates;
2. Zoom in to _50 m_ (BTW I doubt that users understand what _16+_ levels are, so it's better to write _50+ m_ in change logs and so on);
3. Change zoom level from the zoomed out to zoomed in "border" for _50 m_ - area fill will disappear on "zoomed out" border and appear on "zoomed in" border.

[Video](https://user-images.githubusercontent.com/22843816/178195396-1afd7f26-f77d-4a6b-8876-8d6a81238dfc.mp4)

**Comment by pastk** [comment]
> I wanted to create a separate issue but I guess this one has the same roots.
> 

I believe this particular pierses/breakwaters case was fixed already..

**Comment by pastk** [comment]
> Are there any use cases when we don't want this behavior?

E.g. atm we display are fills of "regular" `leisure-garden` and "private" `leisure-garden-residential` identically from z12-.
But might it make sense to lower visibility of private gardens?
On one hand it'll help to keep users' attention focused on more important public features. On the other hand even private features are part of the scenery still and they add a certain character and color (and information) to e.g. neighborhoods with many residential gardens.
WDYT?

**Comment by biodranik** [comment]
@pastk let's compare screenshots!

**Comment by patepelo** [comment]
> > Are there any use cases when we don't want this behavior?
> 
> E.g. atm we display are fills of "regular" `leisure-garden` and "private" `leisure-garden-residential` identically from z12-. But might it make sense to lower visibility of private gardens? On one hand it'll help to keep users' attention focused on more important public features. On the other hand even private features are part of the scenery still and they add a certain character and color (and information) to e.g. neighborhoods with many residential gardens. WDYT?

I agree with you in both options. It may be useful in cases where a user is exploring the map to find public gardens. On the other hand, private gardens are still part of the scenery, and they are useful for orientation and they may be used for some users that have access to them. 

I personally believe that lowering one from the other on a one level zoom difference gives the advantage of exploring-browsing for public gardens with the help of hiding the rest of the private gardens. 

Another approach would be to lower contrast, green on private gardens and/or using icons for public gardens only.
